### ISSUE TITLE
Refactor RestTemplate to WebClient

### ISSUE THREAD

**Issue by shanman190** [openrewrite/rewrite-spring#71]
As of Spring Framework 5.0, RestTemplate has been put into maintenance mode with a recommendation to attempt to use WebClient. A refactoring rule for this would be awesome and should be fairly straightforward.

**Comment by pway99** [comment]
@shanman190 I agree this would be a handy recipe
 For future reference:
[webclient loadbalancer docs](https://spring.getdocs.org/en-US/spring-cloud-docs/spring-cloud-commons/cloud-native-applications/spring-cloud-commons:-common-abstractions/webclinet-loadbalancer-client.html)
[RestTemplate vs WebClient blog](https://ordina-jworks.github.io/rest/2020/10/12/RestTemplate-vs-WebClient.html)

**Comment by shanman190** [comment]
@pway99, yep. I've gone through several `RestTemplate` -> `WebClient` refactors and it's by no means difficult, but it can become tedious. 😆 

Those guides summarize what a recipe may entail pretty well.

**Comment by shanman190** [comment]
Another option here or for a new issue would be to target the new `RestClient` that is coming with Spring Framework 6.1. This component will be reusing the infrastructure that has been built around `RestTemplate` already.

https://spring.io/blog/2023/07/13/new-in-spring-6-1-restclient

**Comment by varaprasadh2** [comment]
do we know an approximate ETA for this?
